.ci/linters/rd/options_doc_check.R

```
@@ -0,0 +1,68 @@
+# Ensure that data.table options in code match documentation
+check_options_documentation = function(rd_file) {
+  if (!grepl("data.table-options", rd_file)) return(invisible())
+  
+  # Find options in R code
+  get_options_from_code = function() {
+    found = character(0)
+    
+    walk_ast = function(expr) {
+      result = character(0)
+      if (is.call(expr) && length(expr) >= 2 && identical(expr[[1]], quote(getOption)) && is.character(expr[[2]]) && startsWith(expr[[2]], "datatable.")) {
+        result = expr[[2]]
+      }
+      if (is.recursive(expr)) {
+        result = c(result, unlist(lapply(expr, walk_ast)))
+      }
+      result
+    }
+    r_files = list.files("R", pattern = "\\.R$", full.names = TRUE)
+    for (file in r_files) {
+      tryCatch({
```

Why `tryCatch()` here? It might hide errors we'd rather see

---

I added them so that it can handle file with syntax error or encoding issue. But you're right I'll remove the tryCatch() so parsing failures doesn't hide